docs-src/api-overview.md

```diff
@@ -79,6 +96,128 @@ BackendPolicy. For more information on what may be configured with this resource
+- **Many-to-one** - Routes can also be bound to more than one Gateway, allowing 
+  a single Route to control application exposure simultaneously across different
+  IPs, load balancers, or networks.
+
+*In summary, Gateways select Routes and Routes also select Gateways. When both 
+select each other, and there are no conflicts between existing bound Routes,
+then a Route will bind to a Gateway. When Routes are bound to a Gateway, it
+means their collective routing rules are configured on the underlying load
+balancers or proxies that are managed by that Gateway. Thus, a Gateway is a
+symbolic representation of a networking data plane that can be configured
+through Routes.*
+
+#### Route Selection
+
+The primary way in which Gateways select Routes (effectively allowing them to
+bind) is via their Route resource labels. Routes are actually bound to specific
+listeners within the Gateway so each listener has a `listener.routes` field
+which selects Routes by different criteria. There are three criteria considered
+for a Gateway to select Routes:
+
+- **Route label** - A Gateway selects Routes via labels that exist on the 
+resource (similar to how Services select Pods via Pod labels).
+- **Route type** - A Gateway listener can only select a single type of Route 
+resource. This could be an HTTPRoute, TCPRoute, or a custom Route type.
+- **Namespace** - A Gateway can also control from which Namespaces Routes can be
+selected via the `namespaces.from` field. It supports three possible values:
+    - `SameNamespace` is the default option. Only Routes in the same namespace as 
+    this Gateway will be selected. 
+    - `All` will select Routes from all Namespaces.
+    - `Selector` means that Routes from a subset of Namespaces selected by a 
+    Namespace label selector will be selected by this Gateway. When `Selector` 
+    is used then the `listeners.routes.namespaces.selector` field can be used to 
+    specify label selectors. This field is not supported with `All` or 
+    `SameNamespace`.
+
+The below Gateway will select all HTTPRoute resources with the `gateway:
+prod-web-gw` across all Namespaces in the cluster.
+
+```
+kind: Gateway
+...
+spec:
+  listeners:  
+  - routes:
+      kind: HTTPRoute
+      selector:
+        matchLabels:
+          gateway: prod-web-gw 
+      namespaces:
+        from: All
+```
+
+#### Gateway Selection
+
+Routes can also select which Gateways they can select. The `gateways.allow`
+field supports three values:
+
+- `All` is the default value if none is specified. This leaves all binding 
+to the Route label and Namespace selectors on the Gateway. 
+- `SameNamespace` only allows this Route to bind with Gateways from the 
+same Namespace.
+- `FromList` allows an explicit list of Gateways to be specifiied that a 
+Route will bind with. `gateways.gatewayRefs` is only supported with this option. 
+
+The following Route can only bind with  `foo-gateway` in the `foo-namespace`.
+
+```yaml
+kind: HTTPRoute
```

Comment:
A minor comment: How about we add a namespace to this route? That will further clarify that the list can contain references to Gateways from a different namespace.

Reply:
good idea - I added that